report/11_Systems-Design.md

```diff
@@ -1 +1,11 @@
-Design of your ITU-MiniTwit systems
+A component digram describing the flow of data through the MiniTwit system itself can be seen in the figure below.  
+
+![Component diagram](./images/component_diagram.png)
+
+The MiniTwit application has two main C# Projects, MiniTwit.Web.App, containing the web app, and MiniTwit.Models containing the repositories for the different types of data handled by the application. The application also has other projects containing the MiniTwitContext connecting to the Database, or containing utility functions, but these are not illustrated in the diagram above. 
+
+The ```MinitwitContext``` (based in EntityFrameworkCore) functions as the database abstraction layer, and allows the repositories to access their relevant data. The repositories then contain the relevant methods for passing the data on in specific formats to the frontend controllers, contained in ```MiniTwit.Web.App```. The ```HomeController``` passes this data on to the views in the Views directory, where they are used when accessed in a browser, while the ```APIcontroller``` instead exposes the API that the simulator uses.
+
+In general, the application is written in C# whenever possible, partially because the team wanted to learn the different aspects of C#, and partially because it made it easier to connect all the parts of the application. Thus, the frontend of the application is written in ASP.NET pages, the security layer is made using ASP.NET Identity, and EntityFrameworkCore is used for the database abstraction layer.
+
+The choice of the general repository structure, was made because the team had worked with the structure in C# before, and thus would be able to implement it quickly, and have a reference point of how it should be set up.
```

Comment:
Why is this sentence here? "general repository structure" is that not fore the repo chapter.
What was the reference? Link it if possible.

Reply:
I guess the real phrase is design pattern, not structure. The repo chapter is about the repository the project is hosted on, as in the Github Repository

Reply:
I would remove most of this sentence then. Or change it to something like:

> This application design takes inspiration from the BDSA course (ref) in which we implemented a similiar structure for seperating application and database logic. By reusing a familiar structure it allowed for faster implementation and therefore more time to focus on DevOps related tasks.